internal/options/alias.go

```diff
@@ -0,0 +1,246 @@
+package options
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/iancoleman/strcase"
+	"github.com/launchdarkly/ld-find-code-refs/internal/validation"
+	"gopkg.in/yaml.v2"
+)
+
+// We're using multiple packages for configuration.
+// TODO: Use spf13/viper to standardize configuration for args, env, and yaml.
+
+type AliasType string
+
+func (a AliasType) IsValid() error {
+	switch a {
+	case Literal, CamelCase, PascalCase, SnakeCase, UpperSnakeCase, KebabCase, DotCase, FilePattern, Command:
+		return nil
+	}
+	return fmt.Errorf("'%s' is not a valid alias type", a)
+}
+
+func (t AliasType) unexpectedFieldErr(field string) error {
+	return fmt.Errorf("unexpected field for %s alias: '%s'", t, field)
+}
+
+func (a Alias) Generate(flag string) ([]string, error) {
+	ret := []string{}
+	switch a.Type {
+	case Literal:
+		ret = a.Flags[flag]
+	case CamelCase:
+		ret = []string{strcase.ToLowerCamel(flag)}
+	case PascalCase:
+		ret = []string{strcase.ToCamel(flag)}
+	case SnakeCase:
+		ret = []string{strcase.ToSnake(flag)}
+	case UpperSnakeCase:
+		ret = []string{strcase.ToScreamingSnake(flag)}
+	case KebabCase:
+		ret = []string{strcase.ToKebab(flag)}
+	case DotCase:
+		ret = []string{strcase.ToDelimited(flag, '.')}
+	case FilePattern:
+		pattern := regexp.MustCompile(strings.ReplaceAll(*a.Pattern, "FLAG_KEY", flag))
+		results := pattern.FindAllStringSubmatch(string(a.FileContents), -1)
+		for _, res := range results {
+			if len(res) > 1 {
+				ret = append(ret, res[1:]...)
+			}
+		}
+	case Command:
+		ctx := context.Background()
+		if a.Timeout != nil && *a.Timeout > 0 {
+			var cancel context.CancelFunc
+			ctx, cancel = context.WithDeadline(ctx, time.Now().Add(time.Second*time.Duration(*a.Timeout)))
+			defer cancel()
+		}
+		tokens := strings.Split(*a.Command, " ")
+		name := tokens[0]
+		args := []string{}
+		if len(tokens) > 1 {
+			args = tokens[1:]
+		}
+		/* #nosec */
+		cmd := exec.CommandContext(ctx, name, args...)
+		cmd.Stdin = strings.NewReader(flag)
+		cmd.Dir = Dir.Value()
+		stdout, err := cmd.Output()
+		if err != nil {
+			return nil, fmt.Errorf("failed to execute alias command: %w", err)
+		}
+		err = json.Unmarshal(stdout, &ret)
+		if err != nil {
+			return nil, fmt.Errorf("could not unmarshal json output of alias command: %w", err)
+		}
+	}
+
+	return ret, nil
+}
+
+const (
+	Literal AliasType = "literal"
+
+	CamelCase      AliasType = "camelCase"
+	PascalCase     AliasType = "pascalCase"
+	SnakeCase      AliasType = "snakeCase"
+	UpperSnakeCase AliasType = "upperSnakeCase"
+	KebabCase      AliasType = "kebabCase"
+	DotCase        AliasType = "dotCase"
+
+	FilePattern AliasType = "filePattern"
+
+	Command AliasType = "command"
+)
+
+// Alias is a catch-all type for alias configurations
+// TODO: can we use OOP?
```

Comment:
I haven't figured out a good way to break this out into an interface type using yaml, but maybe it will be easier with viper